app/src/main/java/org/mifos/mobile/viewModels/HomeViewModel.kt

```diff
@@ -26,21 +26,29 @@ class HomeViewModel @Inject constructor(private val homeRepositoryImp: HomeRepos
 
     private val _homeUiState = MutableStateFlow<HomeUiState>(HomeUiState.Loading)
     val homeUiState: StateFlow<HomeUiState> = _homeUiState
-
+    var errorHandled = false
     fun loadClientAccountDetails() {
         viewModelScope.launch {
             _homeUiState.value = HomeUiState.Loading
-            homeRepositoryImp.clientAccounts().catch {
-                _homeUiState.value = HomeUiState.Error(R.string.error_fetching_accounts)
+
+
+            homeRepositoryImp.clientAccounts().catch { 
+                if (!errorHandled) {
```

Comment:
do we need this condition ? I mean the purpose of catch api itself is to catch any exception that has occurred in flows execution

Reply:
The condition if (!errorHandled) is there to ensure that the error-handling block inside the catch block is executed only if errorHandled is false. This is likely done to prevent redundant error handling and ensure that the error state is set only once.

Reply:
sorry but I didn't understand this explanation. If I am inside the catch block then how is it making error handling redundant ?

Reply:
The condition if (!errorHandled) checks whether the error handling logic inside the catch block should be executed. If errorHandled is true, it means that the error has already been handled, and there is no need to handle it again.
the issue is (the Failed to fetch details dialogue appears constantly multiple times) right....it ensures the snack bar appears  pnly once..

Reply:
@PratyushSingh07 Yeah its working fine even though the if statements are removed....ig there is no issue it was my problem

Reply:
Then add another commit that fixes these changes

Reply:
Done!